beacon/blockchain/init_chain.go

```diff
@@ -71,33 +72,27 @@ func (s *Service) ProcessGenesisData(
 		genesisVersion,
 	)
 	if err != nil {
-		return nil, nil, common.Root{}, common.Root{}, err
+		return nil, nil, common.Root{}, nil, err
 	}
 
 	// After deposits are validated, store the genesis deposits in the deposit store.
 	if err = s.storageBackend.DepositStore().EnqueueDeposits(
 		ctx,
 		genesisData.GetDeposits(),
 	); err != nil {
-		return nil, nil, common.Root{}, common.Root{}, err
+		return nil, nil, common.Root{}, nil, err
 	}
 
 	// Get the genesis beacon block header from the state.
 	genesisHeader, err := genesisState.GetLatestBlockHeader()
 	if err != nil {
-		return nil, nil, common.Root{}, common.Root{}, err
+		return nil, nil, common.Root{}, nil, err
 	}
 
 	// Return the hash tree root of the genesis header after updating the state root in it.
 	// This is similar to how we get the block root.
 	genesisHeader.SetStateRoot(genesisState.HashTreeRoot())
 	genesisBlockRoot := genesisHeader.HashTreeRoot()
 
-	// Get the genesis validators root from the state.
-	genesisValidatorsRoot, err := genesisState.GetGenesisValidatorsRoot()
-	if err != nil {
-		return nil, nil, common.Root{}, common.Root{}, err
-	}
-
-	return validatorUpdates, genesisHeader, genesisValidatorsRoot, genesisBlockRoot, nil
+	return validatorUpdates, genesisHeader, genesisBlockRoot, genesisState, nil
```

Comment:
rather than returning all 3 just return the genesisState because the others can be derived from state.

Reply:
yeah that's the plan to derive header and blockroot from genesis state. Before that refactor I wanted to test few cases, seems not getting correct balances from the genesisState due to which e2e tests are breaking.

Reply:
addressed, just returning genesisState and deriving values from state in `initChainer`